ast/src/analysis/mod.rs

```diff
@@ -0,0 +1,106 @@
+mod display;
+
+use std::collections::BTreeSet;
+
+use num_bigint::BigUint;
+
+use crate::parsed::{
+    asm::{InstructionBodyElement, InstructionParams, RegisterFlag},
+    Expression, Statement,
+};
+
+pub struct RegisterDeclarationStatement {
+    pub start: usize,
```

Comment:
What's start? (here and in other structs)

Reply:
I think it's span information from the parser